test/test_sf_import_c99.cpp

```diff
@@ -341,13 +348,27 @@ void test()
    }
    s   = 8 * std::numeric_limits<T>::epsilon();
    val = 2.5;
-   BOOST_CHECK_CLOSE_FRACTION(asinh(val), T("1.6472311463710957106248586104436196635044144301932365282203100930843983757633104078778420255069424907777006132075516484778755360595913172299093829522950397895699619540523579875476513967578478619028438291006578604823887119907434"), s);
-   BOOST_CHECK_CLOSE_FRACTION(asinh(val + T(0)), T("1.6472311463710957106248586104436196635044144301932365282203100930843983757633104078778420255069424907777006132075516484778755360595913172299093829522950397895699619540523579875476513967578478619028438291006578604823887119907434"), s);
+   {
+      #if defined(TEST_CPP_DOUBLE_FLOAT)
```

Comment:
As above we may have multiple TEST_* macros set.

Reply:
I need to get back to you on this.

Reply:
Hi John (@jzmaddock) thanks for the direction on this.

Fixed in 284547464dfb5d5c859488a144f812b7ab8ead15.

I opted for a new file called `test_traits.hpp` under the assumption that this would be a good place to collect (in the future) all of these kinds of things and reduce redundancies.